es-core/src/resources/TextureData.cpp

```diff
@@ -46,33 +46,22 @@ bool TextureData::initSVGFromMemory(const unsigned char* fileData, size_t length
 
 	NSVGimage* svgImage = nsvgParse(copy, "px", DPI);
 	free(copy);
-	if (!svgImage)
+	if (!svgImage || (svgImage->width == 0) || (svgImage->height == 0))
 	{
 		LOG(LogError) << "Error parsing SVG image.";
 		return false;
 	}
 
 	// We want to rasterise this texture at a specific resolution. If the source size
 	// variables are set then use them otherwise set them from the parsed file
-	if ((mSourceWidth == 0.0f) && (mSourceHeight == 0.0f))
-	{
-		mSourceWidth = svgImage->width;
+	if (mSourceHeight == 0.0f)
 		mSourceHeight = svgImage->height;
-	}
+
+	mSourceWidth = (mSourceHeight * svgImage->width) / svgImage->height;
+
 	mWidth = (size_t)Math::round(mSourceWidth);
 	mHeight = (size_t)Math::round(mSourceHeight);
 
-	if (mWidth == 0)
```

Comment:
Are we opening up any edge case by removing these checks for when the rounding returns zero?

Reply:
No because width is always based on height, which is already not allowed to be 0, the math inside these if's would not return anything more useful than what's already there.